### PR TITLE
metrics: Add a metric to diagnose the write-conflict severity  caused by RCCheckTS isolaion

### PR DESCRIPTION
**Pull request by TonsnakeLin**
### What problem does this PR solve?

Issue Number: Ref #37226

Problem Summary:

### What is changed and how it works?
This PR added a metric `tidb_server_rc_check_ts_conflict_total` whose type is `CounterVec` to count the write-conflict caused by RCCheckTS.
image

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by TonsnakeLin** [comment]
@cfzjywxk @ekexium @MyonKeminta  Thanks for your reviewing!

File: metrics/grafana/tidb.json [hunk 1]
```diff
@@ -3574,6 +3574,93 @@
...
             "alignLevel": null
           }
+        },
+        {
+          "aliasColors": {},
+          "bars": false,
+          "dashLength": 10,
+          "dashes": false,
+          "datasource": "${DS_TEST-CLUSTER}",
+          "description": "TiDB gernerates WriteConflict error count caused by RCCheckTS",
```
Comment (ekexium) [inline metrics/grafana/tidb.json:3584]:
```suggestion
          "description": "The number of WriteConflict errors caused by RCCheckTS",
```

**Review by ekexium — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**